rmtk/vulnerability/common/utils.py

```diff
@@ -1723,3 +1723,104 @@ def convert_array_to_string(array):
+
+    file = open(input_file)
+    data = csv.reader(file)
+    IMLs = []
+    prob_exceedance = []
+    IM_type = []
+    for line in data:
+       if line[0] == 'PGA':
+          IM_type.append(line[0])
+          for value in line[1:]:
+             if isNumber(value):    
+                IMLs.append(float(value))
+      if line[0] == 'Sa':
+         IM_type.append(line[0])
+         for value in line[1:]:
+            if isNumber(value):
+               IMLs.append(float(value))
+      if line[0] == 'PoE':
+          for value in line[1:]:
+             if isNumber(value):
+                prob_exceedance.append(float(value))
+                
+      #Store all the data in the dictionary
+      hazard_curve = {'IMLs':None,'PoE':None,'IM_Type':None}
+      hazard_curve['IMLs'] = IMLs
+      hazard_curve['PoE'] = prob_exceedance
+      hazard_curve['IM_Type'] = IM_type
+      
+      return hazard_curve
+         
+def plot_hazard_curve(hazard_curve):
+#This function plots the hazard curve
+
+   IMLs = hazard_curve['IMLs']
+   PoE = hazard_curve['PoE']
+   plt.plot(IMLs,PoE,color='g',linewidth=2)
+   plt.xlabel(hazard_curve['IM_Type'][0] + ' [g]',fontsize = 10)
+   plt.ylabel('Probability of Exceedance',fontsize = 10)
+
+def read_frag_model(input_file):    
+#This function reads a fragility model and stores it in a dictionary
+
+    damage_states = []
+    model_type = []
+    cent_value = []
+    dispersion = []
+    log_mean = []
+    log_stdev = []
+        
+    file = open(input_file)
+    data = file.readlines()
+    line = data[0]
+    line = line.strip().split(',')
+    model_type = line[1] + ' - ' + line[2]
+    
+    file = open(input_file)
+    data = csv.reader(file)
+    data = [row for row in data]
+    for iline in range(len(data)-1):
+       line = data[iline+1]
+       damage_states.append(line[0])
+       cent_value.append(line[1])
+       dispersion.append(line[2])
+    
+    if model_type == ' median -  dispersion':
+       for ids in range(len(cent_value)):
+          mean = math.log(cent.value[ids])
+          stdev = dispersion[ids]
+          log_mean.append(mean)
+          log_stdev.append(stdev)
+    elif model_type == 'mean of x - cov of x':
+       for ids in range(len(cent_value)):
+          mu = cent.value[ids]
+          cov = dispersion[ids]
+          median = (mu**2)/math.sqrt(mu**2+cov**2)
+          mean = math.log(median)
+          stdev = math.sqrt(math.log(1+(cov**2)/mu**2))
+          log_mean.append(mean)
+          log_stdev.append(stdev)
+    elif model_type == 'mean of ln(x) - st. dev. of ln(x)':
+       for ids in range(len(cent_value)):
+          mean = cent_value[ids]
+          stdev = line[ids]
+          log_mean.append(mean)
+          log_stdev.append(stdev)
+       
+    #Store all the data in the dictionary
+    fragility_model =  {'Damage_states':None,'log_mean':None,'log_stdev':None}
+    fragility_model['Damage_states'] = damage_states
+    fragility_model['log_mean'] = log_mean
```

Comment:
Please check if this is consistent with the other definitions, which use `fragility_model['logmean']`